.github/workflows/tests.yml

```diff
@@ -76,10 +76,12 @@ jobs:
           pytest tests/scenario_tests_async/ --junitxml=reports/test_scenario_async.xml
       - name: Upload test results to Codecov
         if: ${{ !cancelled() }}
-        uses: codecov/test-results-action@47f89e9acb64b76debcd5ea40642d25a4adced9f # v1.1.1
+        uses: codecov/codecov-action@5a1091511ad55cbe89839c7260b706298ca349f7 # v5.5.1
         with:
           directory: ./reports/
+          fail_ci_if_error: true
```

Comment:
👁️‍🗨️ note: We discussed changing this to `false` but that might've been the default used before! I believe changing this action to the recommended one while still keeping this erroring step might be alright for now, but I'm so open to revisiting this!